src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/Telemetry/TelemetryReporter.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common.Telemetry
+{
+    internal static class TelemetryReporter
```

Comment:
This follows a similar way that Roslyn does it. Conversely, we could add a MEF import/export style instead of a static. Would love to get thoughts on this